tests/test_density.py

```diff
@@ -0,0 +1,170 @@
+from argparse import ArgumentParser
+import numpy as np
+import math
+from parcels import Field, Grid, JITParticle, ScipyParticle, Variable
+
+
+def GradientClimber(particle, grid, time, dt):
+    f_lat = dt / 1000. / 1.852 / 60.
+    f_lon = f_lat / math.cos(particle.lat*math.pi/180)
```

Comment:
I can only see a way of declaring `mesh = flat` during construction of a new grid object from netcdf or data. Here, we wish to say that the abstract `K` field, which is used by `GradientClimber` and added to the grid after construction, is on a mesh. I can't see a way to do this?